app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/CustomsAgentController.scala

```diff
@@ -30,28 +29,26 @@ import scala.concurrent.{ExecutionContext, Future}
 class CustomsAgentController @Inject()(
                                         override val controllerComponents: MessagesControllerComponents,
                                         actionProvider: DeclarationJourneyActionProvider,
-                                        formProvider: CustomAgentFormProvider,
                                         repo: DeclarationJourneyRepository,
                                         view: CustomsAgentView,
                                       )(implicit ec: ExecutionContext, appConf: AppConfig) extends DeclarationJourneyUpdateController {
 
-  override val onSubmit: Action[AnyContent] = customsAgentSubmit
-  override val onPageLoad: Action[AnyContent] = customsAgent
-
-  private val form: Form[Boolean] = formProvider()
-
-  private def customsAgent: Action[AnyContent] = actionProvider.journeyAction { implicit request =>
+  val onPageLoad: Action[AnyContent] = actionProvider.journeyAction { implicit request =>
     Ok(view(request.declarationJourney.maybeIsACustomsAgent.fold(form)(asw => form.fill(YesNo.to(asw)))))
   }
 
-  private def customsAgentSubmit: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
-    def onError(): Future[Result] = Future.successful(BadRequest("something WRONG"))
-
-    form.bindFromRequest().fold(_ => onError(), { answer =>
-      repo.upsert(request.declarationJourney.copy(maybeIsACustomsAgent = Some(YesNo.from(answer)))).map { _ =>
-        if (answer) Redirect(routes.SkeletonJourneyController.agentDetails())
-        else Redirect(routes.SkeletonJourneyController.enterEoriNumber())
-      }
-    })
+  val onSubmit: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future successful BadRequest(view(formWithErrors)),
+        isCustomsAgent =>
```

Comment:
I think either we deal in booleans or in YesNos. If its YesNos then better to have a Form[YesNo]. If its a boolean on the page then maybe a boolean on the DeclarationJourney. What do you think?

Reply:
I don't think here is the place to address that issue, this was just refactoring the controller to follow the same pattern as all of the other controllers, and actually return a formWithErrors rather than a page with just "somethingWrong" on it

Reply:
Personally I don't see an issue with booleans. The YesNo thing with from() and to() is just all wrapper nonsense, two options Yes and No maps perfectly to true/false boolean

Reply:
I think YesNo provides a nice way to output 'Yes' or 'No' using messages which can be defined in one place for the entire application so I have a slight preference for YesNo.

Maybe put the 'todo' back into the Form ?

Reply:
For the form on this page it isn't using the messages from YesNo, it's using the ones defined in the inputYesNo.scala.html which just point to site.yes and site.no in messages file

But I'll put the TODO back, no worries

Reply:
Yes but we could change that, which is what the to-do comment is about. Thanks for restoring it.